### PR TITLE
KVStore: Adding statfs and key format fixes to keyvalue store

### PR DESCRIPTION
**Pull request by varadakari**
1. Changing to key format to to preserve the order in backend db
2. Changing the statfs to probe backend if it supports its own statfs

Signed-off-by: Varada Kari <varada.kari@sandisk.com>

### CODE REVIEW

File: src/os/KeyValueStore.cc [hunk 1]
```diff
@@ -592,9 +592,12 @@ KeyValueStore::~KeyValueStore()
 
 int KeyValueStore::statfs(struct statfs *buf)
 {
-  if (::statfs(basedir.c_str(), buf) < 0) {
-    int r = -errno;
-    return r;
+  int r = backend->db->get_statfs(buf);
+  if (r < 0) {
```
Comment (yuyuyu101) [inline src/os/KeyValueStore.cc]: r == -EOPNOTSUPP to avoid accident.

**Comment by varadakari** [comment]
@yuyuyu101 addressed the comment. will squash the commits once the review goes through.

**Comment by liewegas** [comment]
I'd separate out the key changes into a separate PR, or at least separate patches.  @yuyuyu101 should review...

**Comment by varadakari** [comment]
@liewegas Sure sage, It has three changes, will separate them accordingly.

**Comment by varadakari** [comment]
Will split this PR and resubmit again.

**Pull request closed without merging**